core/state/dump.go

```diff
@@ -34,16 +37,76 @@ type DumpAccount struct {
 	Storage  map[string]string `json:"storage"`
 }
 
+// For output in a collected format, as one large map
 type Dump struct {
 	Root     string                 `json:"root"`
 	Accounts map[string]DumpAccount `json:"accounts"`
 }
 
-func (self *StateDB) RawDump() Dump {
-	dump := Dump{
-		Root:     fmt.Sprintf("%x", self.trie.Hash()),
+// DumpAccountFull is the same as DumpAccount but also with address, for standalone printing
+type DumpAccountFull struct {
+	Address  string            `json:"address"`
+	Balance  string            `json:"balance"`
+	Nonce    uint64            `json:"nonce"`
+	Root     string            `json:"root"`
+	CodeHash string            `json:"codeHash"`
+	Code     string            `json:"code"`
+	Storage  map[string]string `json:"storage"`
+}
+
+// For line-by-line json output
+type IterativeDump struct {
```

Comment:
Does this need to be public? Does this even need to exist? Can we have `type iterativeDump *json.Encoder`?

Reply:
I think we do. When doing 
```golang

func (self iterativeDump) onAccount(addr string, account DumpAccount) {
	(*json.Encoder)(self).Encode(&DumpAccountFull{
		addr,
		account.Balance,

```
I'm getting `Invalid receivertype iterativeDump`. [Stack overflow](https://stackoverflow.com/a/44412899) suggests I need to wrap it in a struct.

Reply:
nevermind, works with `type iterativeDump json.Encoder`